packages/block-editor/src/components/block-list/block.js

```diff
@@ -676,14 +706,37 @@ const applyWithSelect = withSelect(
 			getBlockOrder,
 			__unstableGetBlockWithoutInnerBlocks,
 			isNavigationMode,
+			getBlockListSettings,
+			__experimentalGetBlockListSettingsForBlocks,
+			getBlockParents,
 		} = select( 'core/block-editor' );
+
 		const block = __unstableGetBlockWithoutInnerBlocks( clientId );
+
 		const isSelected = isBlockSelected( clientId );
 		const { hasFixedToolbar, focusMode, isRTL } = getSettings();
 		const templateLock = getTemplateLock( rootClientId );
-		const isParentOfSelectedBlock = hasSelectedInnerBlock( clientId, true );
+		const checkDeep = true;
+		const isAncestorOfSelectedBlock = hasSelectedInnerBlock( clientId, checkDeep );
 		const index = getBlockIndex( clientId, rootClientId );
 		const blockOrder = getBlockOrder( rootClientId );
+		const blockParentsClientIds = getBlockParents( clientId );
+		const currentBlockListSettings = getBlockListSettings( clientId );
+
+		// Get Block List Settings for all ancestors of the current Block clientId
+		const ancestorBlockListSettings = __experimentalGetBlockListSettingsForBlocks( blockParentsClientIds );
+
+		// Find the index of the first Block with the `captureDescendantsToolbars` prop defined
+		// This will be the top most ancestor because getBlockParents() returns tree from top -> bottom
+		const topmostAncestorWithCaptureDescendantsToolbarsIndex = findIndex( ancestorBlockListSettings, [ '__experimentalCaptureToolbars', true ] );
+
+		// Boolean to indicate whether current Block has a parent with `captureDescendantsToolbars` set
+		const hasAncestorCapturingToolbars = topmostAncestorWithCaptureDescendantsToolbarsIndex !== -1 ? true : false;
+
+		// Is the *current* Block the one capturing all its descendant toolbars?
+		// If there is no `topmostAncestorWithCaptureDescendantsToolbarsIndex` then
+		// we're at the top of the tree
+		const isCapturingDescendantToolbars = isAncestorOfSelectedBlock && ( currentBlockListSettings && currentBlockListSettings.__experimentalCaptureToolbars ) && ( topmostAncestorWithCaptureDescendantsToolbarsIndex === -1 );
```

Comment:
Minor: In this condition instead of relying on an index comparison we may rely on the flag hasAncestorCapturingToolbars already defined.

Reply:
Thanks. Fixed.